ddtrace/tracer/tracer_test.go

```diff
@@ -1095,6 +1099,25 @@ func decode(p *payload) (spanLists, error) {
 	return traces, err
 }
 
+func (t *dummyTransport) waitFlush(timeout time.Duration) {
+	t.Lock()
+	c := len(t.traces)
+	t.Unlock()
+	for {
+		select {
+		case <-time.After(timeout):
+		default:
+			t.Lock()
+			nc := len(t.traces)
+			t.Unlock()
+			if nc > c {
```

Comment:
This works, but I'm not completely comfortable with it because this is wrapped as `forceFlush` everywhere, so I'm afraid that there might be cases when it could be misleading. What if you have multiple goroutines concurrently finishing spans in a test? The condition may pass for one but not the other.

Is there any reason we aren't doing it like I suggested in https://github.com/DataDog/dd-trace-go/pull/549#discussion_r360280869 ? In my opinion it's much more accurate and "test like". You would do something like: `t.flushAndExpect(5)`. You could even return those traces if you wanted to, because 99% of the time the test needs it.

Yes, you'd have to go to each `forceFlush` call in every test and change it, but that applies even if we keep your current version because `forceFlush` is no longer an appropriate name.

Reply:
Changed to do it how you suggested in #549